salt/modules/keyboard.py

```diff
@@ -63,21 +63,28 @@ def set_sys(layout):
 
         salt '*' keyboard.set_sys dvorak
     """
+    result = None
     if salt.utils.path.which("localectl"):
-        __salt__["cmd.run"]("localectl set-keymap {}".format(layout))
+        result = __salt__["cmd.retcode"]("localectl set-keymap {}".format(layout))
+    #TODO: Make behaviour dependent on outcome of command below
+    # Right now the "result = 0" just mimics the old behaviour, to stop things
+    # breaking, but it's not the right thing to do as it ignores the retcode.
     elif "RedHat" in __grains__["os_family"]:
         __salt__["file.sed"](
             "/etc/sysconfig/keyboard", "^LAYOUT=.*", "LAYOUT={}".format(layout)
         )
+        result = 0
```

Comment:
I see your TODO comment, but I would prefer to not merge this without getting the result of thise command and the other commands per each OS.